frontend/lib/src/components/elements/Video/Video.tsx

```diff
@@ -36,12 +38,27 @@ export default function Video({
   element,
   width,
   endpoints,
+  elementMgr,
 }: VideoProps): ReactElement {
   const videoRef = useRef<HTMLVideoElement>(null)
 
+  const [state] = useState<string>(
+    () => elementMgr.getElementState(element.id) || element.id
+  )
+
   /* Element may contain "url" or "data" property. */
+  const { type, url, startTime, subtitles, endTime, loop, autoplay, muted } =
+    element
+
+  const commitElementValue = (): void => {
```

Comment:
I believe this would need to be a `useCallback` to get rid of the `js_test` warning. But I think since this is anyways just used once I would probably just move the `elementMgr.setElementState(element.id, state)` to the effect below:

```typescript
  useEffect(() => {
    if (element.id && state) {
      elementMgr.setElementState(element.id, state)
    }
  }, [state, element.id, elementMgr])
```

And just having `element.id` as a dependency instead of `element` might be enough here.

Reply:
Same also applies for `Audio.tsx`

Reply:
I was about to add a `useCallback`, based on the suggestion in the warnings, so I'm glad you pointed out that we can simply move the `setElementState` within the effect! Nice 😄 Fixed on 732e722